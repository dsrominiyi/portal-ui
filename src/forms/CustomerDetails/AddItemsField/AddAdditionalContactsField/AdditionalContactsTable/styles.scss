.additional-contacts-table {
  margin-top: 1rem;
}
