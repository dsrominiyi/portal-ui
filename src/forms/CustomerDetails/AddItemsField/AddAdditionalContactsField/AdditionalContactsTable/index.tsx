import React, { FunctionComponent } from 'react';

import TableDisplay from '../../../../../components/TableDisplay';

import { getIndividualFullName } from '../../../../../helpers/customerHelper';

import styles from './styles.scss';

export const AdditionalContactsTable: FunctionComponent<ItemTableProps> = ({
  items: additionalContacts,
  onEdit,
  onDelete
}: ItemTableProps) => {
  const rows = (additionalContacts as AdditionalContact[]).map(contact => {
    const { type, detail } = contact.contactDetails[0];
    return [getIndividualFullName(contact.name), `${type}: ${detail}`, contact.relationship];
  });

  return (
    <TableDisplay
      additionalClasses={[styles.additionalContactsTable]}
      headers={['Name', 'Contact', 'Relationship']}
      rows={rows}
      onEdit={onEdit}
      onDelete={onDelete}
      attributes={{ 'test-id': 'additional-contacts-table' }}
    />
  );
};

AdditionalContactsTable.displayName = 'AdditionalContactsTable';
export default AdditionalContactsTable;
