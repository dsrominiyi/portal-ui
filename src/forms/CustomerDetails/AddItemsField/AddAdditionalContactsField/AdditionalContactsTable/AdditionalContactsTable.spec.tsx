import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import AdditionalContactsTable from '.';

let table: ShallowWrapper;
const additionalContacts = [
  {
    name: { title: 'Mr', forename: 'Robert', middleName: 'Dave', surname: 'Smith' },
    contactDetails: [{ type: ContactType.HomeNumber, detail: '08589392900' }],
    relationship: AdditionalContactRelationship.Sibling
  },
  {
    name: { title: 'Mrs', forename: 'Jane', surname: 'Smith' },
    contactDetails: [{ type: ContactType.WorkEmail, detail: 'jsmith@test.com' }],
    relationship: AdditionalContactRelationship.Spouse
  }
];

describe('AdditionalContactsTable', () => {
  beforeEach(() => {
    table = shallow(
      <AdditionalContactsTable
        items={additionalContacts}
        onEdit={() => null}
        onDelete={() => null}
      />
    );
  });

  test('AdditionalContactsTable matches snapshot', () => {
    expect(table).toMatchSnapshot();
  });
});
