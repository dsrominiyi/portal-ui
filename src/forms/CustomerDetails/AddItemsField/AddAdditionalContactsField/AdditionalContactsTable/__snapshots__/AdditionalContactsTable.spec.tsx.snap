// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdditionalContactsTable AdditionalContactsTable matches snapshot 1`] = `
<TableDisplay
  additionalClasses={
    Array [
      "additionalContactsTable",
    ]
  }
  attributes={
    Object {
      "test-id": "additional-contacts-table",
    }
  }
  headers={
    Array [
      "Name",
      "Contact",
      "Relationship",
    ]
  }
  onDelete={[Function]}
  onEdit={[Function]}
  rows={
    Array [
      Array [
        "Mr Robert Dave Smith",
        "Home Number: 08589392900",
        "Sibling",
      ],
      Array [
        "Mrs Janeundefined Smith",
        "Work Email: jsmith@test.com",
        "Spouse",
      ],
    ]
  }
/>
`;
